tools/validator-client-scripts/src/main.rs

```diff
@@ -0,0 +1,226 @@
+use crate::commands::vesting_undelegate_from_mixnode::vesting_undelegate_from_mixnode;
+use crate::commands::vesting_update_profit_percent::vesting_update_profit_percent;
+use clap::Parser;
+use log::{error, warn};
+use network_defaults::{
+    setup_env,
+    var_names::{BECH32_PREFIX, MIX_DENOM},
+    NymNetworkDetails,
+};
+use validator_client::nymd::{self, AccountId, NymdClient, SigningNymdClient};
+
+mod commands;
+
+// we're always going to be using the signing client
+pub(crate) type Client = validator_client::nymd::NymdClient<SigningNymdClient>;
+
+#[derive(Debug, Parser)]
+pub(crate) enum Command {
+    BondGateway(commands::bond_gateway::Args),
+    BondMixnode(commands::bond_mixnode::Args),
+    ClaimOperatorReward(commands::claim_operator_reward::Args),
+    ClaimDelegatorReward(commands::claim_delegator_reward::Args),
+    CompoundOperatorReward(commands::compound_operator_reward::Args),
+    CompoundDelegatorReward(commands::compound_delegator_reward::Args),
+    Init(commands::init_contract::Args),
+    Migrate(commands::migrate_contract::Args),
+    Upload(commands::upload_contract::Args),
+    UnbondMixnode(commands::unbond_mixnode::Args),
+    UnbondGateway(commands::unbond_gateway::Args),
+    DelegateToMixnode(commands::delegate_to_mixnode::Args),
+    UnDelegateFomMixnode(commands::undelegate_from_mixnode::Args),
+    Send(commands::send::Args),
+    CreateAccount(commands::create_account::Args),
+    UpdateProfitPercent(commands::update_profit_percent::Args),
+    DecodeMixnodeKey(commands::decode_mixnode_key::Args),
+    VestingUpdateProfitPercent(commands::vesting_update_profit_percent::Args),
+    VestingBondGateway(commands::vesting_bond_gateway::Args),
+    VestingBondMixnode(commands::vesting_bond_mixnode::Args),
+    VestingClaimDelegatorRewards(commands::vesting_claim_delegator_reward::Args),
+    VestingClaimOperatorRewards(commands::vesting_claim_operator_reward::Args),
+    VestingCompoundOperatorRewards(commands::vesting_compound_operator_reward::Args),
+    VestingCompoundDelegatorRewards(commands::vesting_compound_delegator_reward::Args),
+    VestingCreateSchedule(commands::vesting_create_schedule::Args),
+    VestingDelegateToMixnode(commands::vesting_delegate_to_mixnode::Args),
+    VestingUnbondGateway(commands::vesting_unbond_gateway::Args),
+    VestingUnbondMixnode(commands::vesting_unbond_mixnode::Args),
+    VestingUndelegateFromMixnode(commands::vesting_undelegate_from_mixnode::Args),
+}
+
+#[derive(Debug, Parser)]
+#[clap(name = "validator-client-scripts")]
+pub(crate) struct Args {
+    #[clap(long)]
+    pub(crate) config_env_file: Option<std::path::PathBuf>,
+
+    #[clap(long)]
+    pub(crate) nymd_url: Option<String>,
+
+    #[clap(long)]
+    pub(crate) mnemonic: Option<bip39::Mnemonic>,
+
+    // it can only be `None` in the case of contract upload or init
+    #[clap(long)]
+    pub(crate) mixnet_contract: Option<AccountId>,
```

Comment:
aha, so the contracts are actually optional unlike what the readme says

Reply:
I would add that they are not used anywhere anymore now, so we could actually remove the contract address arguments altogether.